### ISSUE TITLE
Error finding soundfile path on arm64

### ISSUE THREAD

**Issue by BKLronin** [Nuitka/Nuitka#2577]
Hi! 

trying to compile librosa on arm64 (mac 14.1) With python 3.10. and the following packages (shortended):

```
audioread==3.0.1
lameenc==1.6.3
librosa==0.10.1
Nuitka==1.9.1
numba==0.58.1
numpy==1.26.2
PyAudio==0.2.14
pydub==0.25.1
PySide6-Essentials==6.6.0
scikit-learn==1.3.2
scipy==1.11.4
shiboken6==6.6.0
soundfile==0.12.1
```
Version:

```
1.9.1
Commercial: None
Python: 3.10.11 (v3.10.11:7d4cc5aa85, Apr  4 2023, 19:05:19) [Clang 13.0.0 (clang-1300.0.29.30)]
Flavor: CPython Official
Executable: /Users/bklronin/Development/venv_11/bin/python
OS: Darwin
Arch: arm64
Version C compiler: /usr/bin/clang (clang 15.0.0).
```
Settings:
```
# nuitka-project: --plugin-enable=pyside6
# nuitka-project: --plugin-enable=numpy
# nuitka-project: --include-package=mido
# nuitka-project: --include-package=librosa
# nuitka-project: --follow-imports
# nuitka-project: --noinclude-numba-mode=nofollow
# nuitka-project: --standalone
# nuitka-project: --macos-create-app-bundle
# nuitka-project: --include-data-files=devices.yaml=devices.yaml
# nuitka-project: --include-data-files=config.yaml=config.yaml
# nuitka-project: --include-data-files=sampler_presets.yaml=sampler_presets.yaml

```

Commandline:

` python -m nuitka --standalone --macos-create-app-bundle app.py ` in env

Error: Appears with all python versions from 3.9 to 3.11
```

Nuitka-Inclusion:INFO: Not including forbidden DLL '/Users/bklronin/Development/venv_11/lib/python3.10/site-packages/PySide6/Qt/plugins/imageformats/libqpdf.dylib'.
FATAL: Error, failed to find path @rpath/libsndfile.1.dylib (resolved DLL to /Users/bklronin/Development/venv_11/lib/python3.10/site-packages/_soundfile_data/libsndfile.1.dylib) for /Users/bklronin/Development/venv_11/lib/python3.10/site-packages/_soundfile_data/libsndfile_arm64.dylib from 'soundfile', please report the bug.
```

**Comment by kayhayen** [comment]
I don't see a minimal reproducer.

**Comment by BKLronin** [comment]
Here you go :
```
# nuitka-project: --plugin-enable=numpy
# nuitka-project: --include-package=librosa
# nuitka-project: --follow-imports
# nuitka-project: --noinclude-numba-mode=error
# nuitka-project: --standalone
# nuitka-project: --macos-create-app-bundle

import librosa
import numpy as np

# Generate a random audio array
sr = 22050  # Sample rate
duration = 3.0  # Duration in seconds
t = np.arange(0, duration, 1.0 / sr)
audio_signal = np.random.randn(len(t))

# Compute MFCC features
mfccs = librosa.feature.mfcc(y=audio_signal, sr=sr, n_mfcc=13)
```
[librosa_minimal.py.zip](https://github.com/Nuitka/Nuitka/files/13466644/librosa_minimal.py.zip)

**Comment by kayhayen** [comment]
The include package is not going to do you much good, it ought to not be necessary or else should be fixed separately.

**Comment by BKLronin** [comment]
Yeah thats a artifact from wildly experimenting.

**Comment by BKLronin** [comment]
I removed all of the NUmba decorators and ended up at the same place as when using 
`# nuitka-project: --noinclude-numba-mode=nofollow`
Then the original error from the post shows up. 
`FATAL: Error, failed to find path @rpath/libsndfile.1.dylib (resolved DLL to /Users/bklronin/Development/venv_11/lib/python3.10/site-packages/_soundfile_data/libsndfile.1.dylib) for /Users/bklronin/Development/venv_11/lib/python3.10/site-packages/_soundfile_data/libsndfile_arm64.dylib from 'soundfile', please report the bug.`

The soundfile `libsndfile_arm64.dylib` is in place and working. Maybe really just a path formatting error etc

**Comment by BKLronin** [comment]
Yep, when I copy and rename the `libsndfile_arm64.dylib` to the requested name it compiles.

So probably the name generation in here for the arrches? `standard.nuitka-package.config yaml`

```
- module-name: 'soundfile'
  dlls:
    - from_filenames:
        relative_path: '_soundfile_data'
        prefixes:
          - 'libsndfile*64bit'
      when: 'win32 and arch_amd64'
    - from_filenames:
        relative_path: '_soundfile_data'
        prefixes:
          - 'libsndfile*32bit'
      when: 'win32 and arch_x86'
    - from_filenames:
        relative_path: '_soundfile_data'
        prefixes:
          - 'libsndfile'
      when: 'macos'
```

**Comment by BKLronin** [comment]
After that I end up at:

Now caused by pyaudio:

`FATAL:     Error, problem with dependency scan of '/Users/bklronin/Development/venv_11/lib/python3.10/site-packages/pyaudio/_portaudio.cpython-310-darwin.so' with '/opt/homebrew/opt/portaudio/lib/libportaudio.2.dylib' please report the bug.`

Seems similar to the existing issue.

When not using the minimal_librosa test.

**Comment by BKLronin** [comment]
File is in place but symlinked to /opt/homebrew/Cellar/portaudio/19.7.0/lib from
/opt/homebrew/opt/portaudio/lib/libportaudio.2.dylib

End of latein

**Comment by kayhayen** [comment]
Homebrew is not the best supported, I only ever try much with CPython, but it could be generally wrong with the DLL name indeed, although I believe it worked. I will try and cover it with something in Nuitka-Watch I guess. It may also be that `librosa` is still an unresolved issue generally. I do recall using it occasionally with success, but maybe not on all platforms yet, the numba issue is heavy. I do however think that there was something that meant that librosa doesn't automatically use numba at all. I will look into it, once the 1.9 release woes are over.

**Comment by BKLronin** [comment]
That would be great. I look into nuitka on every realease cycle as it would fix quite a few problems and enable some planned features. 

I havent found anythin to disable numba except removing all the decorators by "hand" and that seemed to work. At least I could still use certain elements like hpss, mfcc, utils etc.

The `PyAudio` (-> portaudio) issue seems seperate as it is now a requirement of `librosa` but used for IO like `sounddevice`. This is kinda strange as I think it compiled Pyaudio as it should the last time when the code was simpler. 

Should have made a new issue for that probably?

Thank you very much.

**Comment by kayhayen** [comment]
You know, I never fix things by hand, if necessary, I do it with anti-bloat changes, but so far I believe this kind of JIT was at some point generally disabled, but re-enabled later, but in principle works in many programs. I think I regularly compile with librosa, but I am not sure of the versions. If there is anything with pyaudio or whatever package, yes, make a separate issue out of it.

**Comment by kayhayen** [comment]
If you want to help, it would be better to spend you time maybe in this way:

If you are available to do a PR, this can be fixed by adding to the yaml configuration, described here: [https://nuitka.net/doc/nuitka-package-config.html](https://nuitka.net/doc/nuitka-package-config.html)

And here is more of a walk through style post: [https://nuitka.net/posts/nuitka-package-config-kickoff.html](https://nuitka.net/posts/nuitka-package-config-kickoff.html)

Here are a example PRs for other packages:

[#2030](https://github.com/Nuitka/Nuitka/pull/2030) 
[#1993](https://github.com/Nuitka/Nuitka/pull/1993)

The idea being that you can yourself add missing DLLs, etc. and maybe even fight the need for the JIT there. However, I feel like there is an environment variable to disable it or something like it, one that Nuitka enforced in the past, but that was going too far, since it actually is supposed to be usable for some things.

**Comment by BKLronin** [comment]
Thanks, I tried it with the kickoff method specifically for the `soundfile` dependency of `librosa` on mac os arm64.

```
- module-name: 'soundfile'
  dlls:
    - from_filenames:
        relative_path: '_soundfile_data'
        prefixes:
          - 'libsndfile*64bit'
      when: 'win32 and arch_amd64'
    - from_filenames:
        relative_path: '_soundfile_data'
        prefixes:
          - 'libsndfile*32bit'
      when: 'win32 and arch_x86'
    - from_filenames:
        relative_path: '_soundfile_data'
        prefixes:
          - 'libsndfile_arm64'
      when: 'macos'
```

I tried multiple variations in this section of the default.yaml and in the error message it even seems correct while ignoring my changes. 

`FATAL: Error, failed to find path @rpath/libsndfile.1.dylib (resolved DLL to /Users/bklronin/Development/venv_11/lib/python3.10/site-packages/_soundfile_data/libsndfile.1.dylib) for /Users/bklronin/Development/venv_11/lib/python3.10/site-packages/_soundfile_data/libsndfile_arm64.dylib from 'soundfile', please report the bug.`

So the thing is, the file is in place and the path seems correct but he is not recognizing the `libsndfile_arm64.dylib` and the fix was to just copy and rename it to `libsndfile.1.dylib`. 

He definitely wants his libsndfile.1.dylib :D

**Comment by kayhayen** [comment]
Ah, the yaml can add instructions to include DLLs, it cannot fix cases, where dependencies fail to resolve. This resolution happens in the Python installation, not in the dist folder, so including more or less is not going to change it. It seems the `@rpath` attempt to resolve fails with `(resolved DLL to /Users/bklronin/Development/venv_11/lib/python3.10/site-packages/_soundfile_data/libsndfile.1.dylib) ` and the question to you would be where exactly *is* it if not there?

**Comment by BKLronin** [comment]
Its a nice system though. 

Its exactly there it just has the wrong name. 

Its exactly the case that you have defined in the `dlldependenciesmacos.py` file with the above code but it doesnt seem to work in this case maybe because of `arm64`.



> The above case seems to be handled by this code already but still throws the error.
> 
> ```
> # Some extension modules seem to reference themselves by a different
>         # extension module name, so use that if it exists.
>         if not os.path.exists(resolved_path) and python_version >= 0x300:
>             so_suffixes = getSharedLibrarySuffixes()[:-1]
> 
>             specific_suffix = so_suffixes[0]
>             abi_suffix = so_suffixes[1]
> 
>             if resolved_path.endswith(specific_suffix):
>                 candidate = resolved_path[: -len(specific_suffix)] + abi_suffix
>             elif resolved_path.endswith(abi_suffix):
>                 candidate = resolved_path[: -len(specific_suffix)] + abi_suffix
>             else:
>                 candidate = None
> 
>             if candidate is not None and os.path.exists(candidate):
>                 resolved_path = candidate
> ```

**Comment by Steven-Sanseau** [comment]
Hi, i have the same error with soundfile and rpath

```console
Nuitka-Plugins:INFO: anti-bloat: Not including '_bisect' automatically in order to avoid bloat, but this may cause: may slow down by using fallback implementation.
[0] Nuitka-Plugins:INFO: anti-bloat: Not including 'matplotlib' automatically in order to avoid bloat, but this may cause: plotting will lack matplotlib.
[0] Nuitka-Plugins:INFO: anti-bloat: Not including '_json' automatically in order to avoid bloat, but this may cause: may slow down by using fallback implementation.
[0] Nuitka:INFO: Completed Python level compilation and optimization.
[0] Nuitka:INFO: Generating source code for C backend compiler.
[0] Nuitka:INFO: Running data composer tool for optimal constant value handling.
[0] Nuitka:INFO: Running C compilation via Scons.
[0] Nuitka-Scons:INFO: Backend C compiler: clang (clang 15.0.0).
[0] Nuitka-Scons:INFO: Backend linking program with 598 files (no progress information available for this stage).
[0] Nuitka-Scons:INFO: Compiled 598 C files using ccache.
[0] Nuitka-Scons:INFO: Cached C files (using ccache) with result 'cache hit': 598
[0] Nuitka-Plugins:INFO: dll-files: Found 1 file DLLs from sounddevice installation.
[0] Nuitka-Plugins:INFO: dll-files: Found 1 file DLLs from soundfile installation.
[0] FATAL: Error, failed to find path @rpath/libsndfile.1.dylib (resolved DLL to /Users/steven/Workspace/.venv/lib/python3.11/site-packages/_soundfile_data/libsndfile.1.dylib) for /Users/steven/Workspace/.venv/lib/python3.11/site-packages/_soundfile_data/libsndfile_arm64.dylib from 'soundfile', please report the bug.
[0] yarn workspace mia-audio build exited with code 1
```

```console
## Build command

poetry run python -m nuitka server.py --follow-imports --standalone
```
![Screenshot 2023-12-05 at 09 18 54](https://github.com/Nuitka/Nuitka/assets/2778764/d268e0d3-966c-41fd-8a20-23b54ca376e4)

I'm also on mac m2.

```console
poetry run python -m nuitka --version
1.9.3
Commercial: None
Python: 3.11.6 (main, Oct  2 2023, 13:45:54) [Clang 15.0.0 (clang-1500.0.40.1)]
Flavor: Homebrew Python
Executable: /Users/Workspace/.venv/bin/python
OS: Darwin
Arch: arm64
Version C compiler: /usr/bin/clang (clang 15.0.0).

```

Thanks for your help

**Comment by BKLronin** [comment]
Hi!

You can copy and rename the `libsndfile_arm64.dylib` to `libsndfile.1.dylib` as a workaround. In the same folder.

**Comment by BKLronin** [comment]
That would be nice. Those libs are hard to substitute.

**Comment by kayhayen** [comment]
So, that `ImportError` was actually a bug in the lazy loader being parsed too late, only after its first user completed, and then never became resolved, but in that case, it is actually needed to make the import visible. So that's fixed and coming to 2.2.1 separately.

**Comment by kayhayen** [comment]
Then, the numba module `numba.np.ufunc.omppool` fails to find libomp on my system, which is normal, but Nuitka is crying out about it. Apparently numba handles this internally to fail to load that module, but needed to list it as an acceptable missing DLL for `numba`, since we are not tolerant by default to catch bugs. This can probably come to 2.2.1 separately as well. I had noticed this issue before, but I thought it's using the one from `skimage` maybe in a more complex case, but skimage is not loaded in this example, so that was the wrong solution.

**Comment by kayhayen** [comment]
Now I am facing this one:

> FATAL: Error, failed to find path @rpath/libsndfile.1.dylib (resolved DLL to /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/_soundfile_data/libsndfile.1.dylib) for /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/_soundfile_data/libsndfile_arm64.dylib from 'soundfile', please report the bug.

**Comment by kayhayen** [comment]
So, the path that aims to resolve this doesn't know how to gracefully handle the `.1` at the same time. It would accept the arm64 file only if it had that, too. Seems, we might need to factor that speculative DLL name striping out, and re-use it there too.

**Comment by BKLronin** [comment]
Those .1 numbers are version numbers? So it is in doubt if its the same version if its missing?

**Comment by kayhayen** [comment]
It happens a lot that DLLs depends on versioned names of themselves and vice versa, and so does this DLL too. But i also needed to add the `_arm64` handling, and now it works, well except of course that for `stencil` of `numba` there is a need for the bytecode of the function, and that's something I now will need to add support for.

For compiled functions, we normally only put a pseudo bytecode, and `numba` errors out accessing it for "interpretation", so we either create an uncompiled function object here instead, or we attach the original bytecode to the compiled function, I don't know yet, and I somehow believe we might need or want both eventually. Some people only want to look at the bytecode to know things about code, some want to run it, some the source, etc. so that is where the JIT stuff starts. I like this example code as it is very self contained.

The "soundfile" specific bugs, the import bugs, etc. are all resolved, and will be in 2.2.1 likely, since they all have some value of their own, but for 2.3 we will need to work harder to get this JIT under control. Most numba stuff works for now by disabling the JIT, which of course, too, is not what we want long term, but we can first focus on this `stencil` JIT thing until we get librosa back to business.

**Comment by kayhayen** [comment]
Thanks for your report, this is worked on the factory branch, which is a development version under rapid development. You can try it out by going here: https://nuitka.net/doc/factory.html 

Feedback on whether this is working is very welcome. Please do not share plans to do it, but rather confirmations or denials of it working.

**Comment by BKLronin** [comment]
Result when running it on my full application:

`(venv_nuit11) bklronin@Mini-von-Thomas Development % python -m pip install -U --force-reinstall "https://github.com/Nuitka/Nuitka/archive/factory.zip"                                                     
Collecting https://github.com/Nuitka/Nuitka/archive/factory.zip
  Downloading https://github.com/Nuitka/Nuitka/archive/factory.zip
     - 5.7 MB 7.8 MB/s 0:00:00
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting ordered-set>=4.1.0 (from Nuitka==2.3rc1)
  Downloading ordered_set-4.1.0-py3-none-any.whl.metadata (5.3 kB)
Collecting zstandard>=0.15 (from Nuitka==2.3rc1)
  Downloading zstandard-0.22.0-cp311-cp311-macosx_11_0_arm64.whl.metadata (2.9 kB)
Downloading ordered_set-4.1.0-py3-none-any.whl (7.6 kB)
Downloading zstandard-0.22.0-cp311-cp311-macosx_11_0_arm64.whl (703 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 703.4/703.4 kB 7.9 MB/s eta 0:00:00
Building wheels for collected packages: Nuitka
  Building wheel for Nuitka (pyproject.toml) ... done
  Created wheel for Nuitka: filename=Nuitka-2.3rc1-cp311-cp311-macosx_10_9_universal2.whl size=3278437 sha256=f0c328f68f5df3bd018c4a1a42a558b12b8ec75c545e6efcf2681e402a4c0c67
  Stored in directory: /private/var/folders/kg/zm48w_r96yb68mlbzvb9gtq40000gn/T/pip-ephem-wheel-cache-prog6a2v/wheels/ee/2a/3a/cb4da106665c3f2353bea25a22a4faeb739c2e24e83a0bda26
Successfully built Nuitka
Installing collected packages: zstandard, ordered-set, Nuitka
  Attempting uninstall: zstandard
    Found existing installation: zstandard 0.22.0
    Uninstalling zstandard-0.22.0:
      Successfully uninstalled zstandard-0.22.0
  Attempting uninstall: ordered-set
    Found existing installation: ordered-set 4.1.0
    Uninstalling ordered-set-4.1.0:
      Successfully uninstalled ordered-set-4.1.0
Successfully installed Nuitka-2.3rc1 ordered-set-4.1.0 zstandard-0.22.0
(venv_nuit11) bklronin@Mini-von-Thomas Development % python -m nuitka --standalone --macos-create-app-bundle app.py                                                                             
Nuitka-Options: Used command line options: --standalone --macos-create-app-bundle app.py
FATAL: Error, file 'app.py' is not found.
(venv_nuit11) bklronin@Mini-von-Thomas Development % python -m nuitka --standalone --macos-create-app-bundle ubridge-pro.py 
Nuitka-Options: Used command line options: --plugin-enable=pyside6 --plugin-enable=numpy --include-package=mido --standalone --macos-create-app-bundle --include-data-files=devices.yaml=devices.yaml --include-data-files=config.yaml=config.yaml
Nuitka-Options: --include-data-files=sampler_presets.yaml=sampler_presets.yaml --standalone --macos-create-app-bundle ubridge-pro.py
Nuitka-Plugins:WARNING: numpy: This plugin has been deprecated, do not enable it anymore.
Nuitka: Starting Python compilation with Nuitka '2.3rc1' on Python '3.11' commercial grade 'not installed'.
Nuitka-Plugins:WARNING: options-nanny: Using module 'numba' (version 0.58.1) with incomplete support due to condition 'not use_numba and standalone': Numba is not yet fully working with Nuitka standalone, try to use '--noinclude-numba-mode=nofollow', but it may fail.
Nuitka-Plugins:WARNING: options-nanny: Module has parameter: Numba JIT is disabled by default in standalone mode, make a choice explicit with '--module-parameter=numba-disable-jit=yes|no'
Nuitka-Plugins:options-nanny: Note, when using 'PySide6', consider using '--disable-console' option. Otherwise high resolution will not be available and a terminal window will open. However for debugging, terminal output is the easiest way to see informative traceback and error information, so
Nuitka-Plugins:options-nanny: delay this until your program is working and remove once you find it non-working, and use '--enable-console' to make it explicit and not see this message.
Nuitka-Plugins:anti-bloat: Not including '_bisect' automatically in order to avoid bloat, but this may cause: may slow down by using fallback implementation.
Nuitka-Plugins:anti-bloat: Not including '_json' automatically in order to avoid bloat, but this may cause: may slow down by using fallback implementation.
Nuitka: Completed Python level compilation and optimization.
Nuitka: Generating source code for C backend compiler.
Nuitka: Running data composer tool for optimal constant value handling.                                                       
Nuitka: Running C compilation via Scons.
Nuitka-Scons: Backend C compiler: clang (clang 15.0.0).
Nuitka-Scons: Backend linking program with 1337 files (no progress information available for this stage).
Nuitka-Scons: Compiled 1337 C files using ccache.
Nuitka-Scons: Cached C files (using ccache) with result 'cache miss': 1337
Nuitka-Plugins:pyside6: Including Qt plugins 'iconengines,imageformats,platforms,styles,tls' below 'PySide6/qt-plugins'.
Nuitka-Plugins:dll-files: Found 1 file DLLs from llvmlite.binding.ffi installation.
Nuitka-Plugins:dll-files: Found 2 files DLLs from soundfile installation.
Nuitka-Inclusion: Not including due to forbidden DLL '/Volumes/Data_drive/User/bklronin/Development/venv_nuit11/lib/python3.11/site-packages/numba/np/ufunc/omppool.cpython-311-darwin.so'.
Detecting used DLLs: 94.6%|███████████████████████▋ | 264/279, PySide6/qt-plugins/imageformats/libqpdf.dylibTraceback (most recent call last):
  File "/Volumes/Data_drive/User/bklronin/Development/venv_nuit11/lib/python3.11/site-packages/nuitka/__main__.py", line 161, in <module>
    main()
  File "/Volumes/Data_drive/User/bklronin/Development/venv_nuit11/lib/python3.11/site-packages/nuitka/__main__.py", line 147, in main
    MainControl.main()
  File "/Volumes/Data_drive/User/bklronin/Development/venv_nuit11/lib/python3.11/site-packages/nuitka/MainControl.py", line 1124, in main
    _main()
  File "/Volumes/Data_drive/User/bklronin/Development/venv_nuit11/lib/python3.11/site-packages/nuitka/MainControl.py", line 1016, in _main
    detectUsedDLLs(
  File "/Volumes/Data_drive/User/bklronin/Development/venv_nuit11/lib/python3.11/site-packages/nuitka/freezer/Standalone.py", line 328, in detectUsedDLLs
    _detectUsedDLLs(
  File "/Volumes/Data_drive/User/bklronin/Development/venv_nuit11/lib/python3.11/site-packages/nuitka/freezer/Standalone.py", line 244, in _detectUsedDLLs
    used_dlls = _detectBinaryDLLs(
                ^^^^^^^^^^^^^^^^^^
  File "/Volumes/Data_drive/User/bklronin/Development/venv_nuit11/lib/python3.11/site-packages/nuitka/freezer/Standalone.py", line 140, in _detectBinaryDLLs
    return detectBinaryPathDLLsMacOS(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Volumes/Data_drive/User/bklronin/Development/venv_nuit11/lib/python3.11/site-packages/nuitka/freezer/DllDependenciesMacOS.py", line 81, in detectBinaryPathDLLsMacOS
    had_self, resolved_result = _resolveBinaryPathDLLsMacOS(
                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Volumes/Data_drive/User/bklronin/Development/venv_nuit11/lib/python3.11/site-packages/nuitka/freezer/DllDependenciesMacOS.py", line 272, in _resolveBinaryPathDLLsMacOS
    acceptable, plugin_name = Plugins.isAcceptableMissingDLL(
    ^^^^^^^^^^^^^^^^^^^^^^^
TypeError: cannot unpack non-iterable bool object
Nuitka-Reports: Compilation crash report written to file 'nuitka-crash-report.xml'. Please include it in your bug report.
Detecting used DLLs: 94.6%|███████████████████████▋ | 264/279, PySide6/qt-plugins/imageformats/libqpdf.dylib%     `

[nuitka-crash-report.xml.zip](https://github.com/Nuitka/Nuitka/files/15193427/nuitka-crash-report.xml.zip)

**Comment by kayhayen** [comment]
That's a valid finding, technically unrelated though, I will fix that too.

**Comment by BKLronin** [comment]
I was just too curious if the whole thing would compile and too lazy to copy and paste the minimal app.py :D

Exciting to watch.

**Comment by kayhayen** [comment]
So, the fixes of 2.2.1 are out, the numba work got stuck with me dealing with a bunch of regressions coming out of this work, but I hope to make it for 2.3 still.

**Comment by BKLronin** [comment]
Very well, 

Iḿ probably switching to nuitka with the next major release of my app ,earliest.

Let me know if  you need testing.

**Comment by Ivorforce** [comment]
I'm getting what I assume is the same error (macOS 12) with a slightly different flavour. This is notably using Nuitka 2.4rc8 (40529e7) from the factory branch, which I'm assuming includes the aforementioned fixes:

```
[...]
Nuitka-Scons: Compiled 15006 C files using ccache.
Nuitka-Scons: Cached C files (using ccache) with result 'cache miss': 15006
Nuitka-Plugins:dll-files: Found 1 file DLLs from kiwisolver installation.
Nuitka-Plugins:dll-files: Found 1 file DLLs from soundfile installation.
FATAL:     Error, problem with dependency scan of '~/work/ECG-Viewer/ECG-Viewer/.venv/lib/python3.9/site-packages/_soundfile_data/libsndfile_x86_64.dylib' with '/usr/local/lib/libsndfile.1.dylib' please report the bug.
Error: Process completed with exit code 1.
```

The full output / run can be found here:
https://github.com/Ivorforce/ECG-Viewer/actions/runs/9748479421/job/26903542049

Locally I'm experiencing the same issue (macOS 14, intel mac):

```
[...]
Nuitka-Scons: Compiled 15006 C files using ccache.
Nuitka-Scons: Cached C files (using ccache) with result 'cache miss': 15006
Nuitka-Plugins:dll-files: Found 1 file DLLs from kiwisolver installation.
Nuitka-Plugins:dll-files: Found 1 file DLLs from soundfile installation.
FATAL:     Error, problem with dependency scan of '~/Library/Caches/pypoetry/virtualenvs/ecg-viewer-ZneG1Rn7-py3.9/lib/python3.9/site-packages/_soundfile_data/libsndfile_x86_64.dylib' with '/usr/local/lib/libsndfile.1.dylib' please report the bug.
```

**Comment by kayhayen** [comment]
So this is delayed until 2.5 due to more pressing issues, the numba enhanced support will be needed for librosa no doubt and more things in the future I suspect.
